packages/execution-environments/src/common/BaseSnapExecutor.ts

```diff
@@ -62,19 +74,19 @@ export class BaseSnapExecutor {
 
     this.methods = getCommandMethodImplementations(
       this.startSnap.bind(this),
-      (target, origin, request) => {
+      (target, handlerName, args) => {
         const data = this.snapData.get(target);
-        if (data?.exports?.onRpcRequest === undefined) {
+        if (data?.exports?.[handlerName] === undefined) {
           throw new Error(
-            `No onRpcRequest handler exported for snap "${target}`,
+            `No ${handlerName} handler exported for snap "${target}`,
           );
         }
 
         // We're capturing the handler in case someone modifies the data object before the call
-        const handler = data.exports.onRpcRequest;
-        return this.executeInSnapContext(target, () =>
-          handler({ origin, request }),
-        );
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const handler = data.exports[handlerName]!;
+        // @todo fix type
+        return this.executeInSnapContext(target, () => handler(args as any));
```

Comment:
Does the `SnapRpcHookArgs` type not work here?

Reply:
I think the problem is that TypeScript doesnt know which of the args types to use so it complains because not all types fit in all handlers. I wasn't sure how to deal with it at the time 🤔